File: vipclaw-admin/src/main/java/com/vipamp/vipclaw/admin/exception/BizException.java
package com.vipamp.vipclaw.admin.exception;

import lombok.Getter;

/**
 * 业务异常类
 */
@Getter
public class BizException extends RuntimeException {

    private final Integer code;

    public BizException(String message) {
        super(message);
        this.code = 400;
    }

    public BizException(Integer code, String message) {
        super(message);
        this.code = code;
    }

    public BizException(String message, Throwable cause) {
        super(message, cause);
        this.code = 400;
    }

    public BizException(Integer code, String message, Throwable cause) {
        super(message, cause);
        this.code = code;
    }
}

File: harnax-admin/src/main/kotlin/com/agnetix/harnax/admin/exception/BizException.kt
package com.agnetix.harnax.admin.exception

/**
 * Business exception class
 */
class BizException : RuntimeException {

    val code: Int

    constructor(message: String) : super(message) {
        this.code = 400
    }

    constructor(code: Int, message: String) : super(message) {
        this.code = code
    }

    constructor(message: String, cause: Throwable) : super(message, cause) {
        this.code = 400
    }

    constructor(code: Int, message: String, cause: Throwable) : super(message, cause) {
        this.code = code
    }
}
